std/stdio.d

```diff
@@ -3186,14 +3186,29 @@ void main()
 /// Get the size of the file, ulong.max if file is not searchable, but still throws if an actual error occurs.
     @property ulong size() @safe
     {
-        import std.exception : collectException;
+        import std.exception : enforce, collectException;
+
+        enforce(isOpen, "Attempting to seek() in an unopened file");
 
         ulong pos = void;
-        if (collectException(pos = tell)) return ulong.max;
+        if (collectException(pos = tell))
+            return ulong.max;
+
+        // Unfortunately there doesn't seem to be a reliable way to
+        // find out if a file is seekable without actually attempting
+        // to seek in it. Even seek(0, SEEK_CUR) will succeed in
+        // unseekable files.
+        if (collectException(seek(0, SEEK_END)))
+            return ulong.max;
+
         scope(exit) seek(pos);
-        seek(0, SEEK_END);
         return tell;
     }
+
+    @system unittest // Issue 17228
+    {
+        assert(stdin.size == ulong.max);
```

Comment:
This seems to throw an error on some systems (interestingly, only 32-bit ones, maybe a hint there).

Reply:
Hm... are we sure stdin is always not a file? If you do `testprogram < input`, this may well succeed.

Reply:
But I suppose in this context, it shouldn't be. It would probably be difficult to set up a true non-seekable file portably and reliably.